### ISSUE TITLE
[Accessibility]: Branch/Tag/Commit: 'Find...' input placeholders aren't descriptive

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35477]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34417

### Problem description

<img width="572" alt="image" src="https://user-images.githubusercontent.com/9516420/168598522-e72dcba4-045d-4799-b218-ce9c0225d6f0.png">

Each tab has the "Find..." input placeholder, this is not descriptive and is difficult for screen readers to understand the purpose of the input.

### Expected behavior

We should add an additional `aria-label` that is more descriptive for example:

> Find branches/

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
